lib/rules/no-attrs-in-components.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const utils = require('../utils/utils');
+
+const MESSAGE = 'Do not use this.attrs';
+
+//------------------------------------------------------------------------------
+// General rule - Don't use this.attrs
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Disallow usage of this.attrs in components',
+      category: 'General',
+      recommended: true
+    },
+    fixable: null,
+  },
+
+  create(context) {
+    return {
+      MemberExpression(node) {
```

Comment:
We have an util for checking that: https://github.com/ember-cli/eslint-plugin-ember/blob/master/lib/rules/order-in-components.js#L40

You should then add `CallExpression:exit` selector, store globally the found member expressions and then check if any of these exist inside that CallExpression. For the reference I wrote a similar check here: https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/no-side-effects-in-computed-properties.js#L39